Feature/us14148

bwf93 opened the pull request:
The new command can be run with:
sudo tools/serviced-storage/serviced-storage sync -p /tmp/outdata/ --source /project/europa/var/serviced/volumes/ -o dm.thinpooldev=/dev/mapper/serviced-serviced--pool

jplouis commented:
retest this

bwf93 commented:
retest this since jenkins jobs fail for random reasons

jafcrocker commented on tools/serviced-storage/volume.go:
```diff
@@ -41,29 +42,83 @@ type VolumeRemove struct {
+type VolumeSync struct {
+	Args struct {
+		DestinationPath flags.Filename `description:"Path of the destionation" long:"destination-path" short:"p" required:"yes" `
+		SourcePath flags.Filename      `description:"Path of the source driver" long:"source" short:"s" required:"yes"`
```
I think these should be positional args, not options.

bwf93 commented:
Changed to use positional arguments for the source and destination and to optionally init the driver.  Also added logging for each volume that is synced.

New example run
sudo tools/serviced-storage/serviced-storage sync /project/europa/var/serviced/volumes/ /tmp/outdata  -c  --type devicemapper -o dm.thinpooldev=/dev/mapper/serviced-serviced--pool

iancmcc commented on tools/serviced-storage/volume.go:
```diff
@@ -44,6 +47,91 @@ type VolumeRemove struct {
+		}
+	}
+	destinationDirectory := GetDefaultDriver(destinationPath)
+	logger.Info("Getting driver for destination")
```
Make this debug-level. It should nearly never fail.

iancmcc commented on tools/serviced-storage/volume.go:
```diff
@@ -44,6 +47,91 @@ type VolumeRemove struct {
+		logger = logger.WithFields(log.Fields{
+			"type":c.Type,
+		})
+		logger.Info("Determining driver type for destination")
```
Make this debug-level.

iancmcc commented on tools/serviced-storage/volume.go:
```diff
@@ -44,6 +47,91 @@ type VolumeRemove struct {
+		logger.Fatal(err)
+	}
+	sourceDirectory := GetDefaultDriver(string(sourcePath))
+	logger.Info("Getting driver for source")
```
And this.
